### PR TITLE
Issue 66 - Pull request for master.

### PR DESCRIPTION
**Pull request by bnandaku**
Please refer to issue #66.

### CODE REVIEW

File: public/javascripts/controllers/widget.portal.js [hunk 1]
```diff
@@ -115,8 +115,10 @@ define([
 
     $scope.generateWidgetCode = function () {
       console.log("--> Generating Output Code...");
+        //Good Notification
 
       if ($scope.widget.basic_auth !== '1' && !($scope.validAuthUri || false)) {
+        //Bad Notification
```
Comment (san983) [inline public/javascripts/controllers/widget.portal.js]: Please add a notification here with the same text as below

File: public/javascripts/controllers/widget.portal.js [hunk 2]
```diff
@@ -115,8 +115,10 @@ define([
 
     $scope.generateWidgetCode = function () {
       console.log("--> Generating Output Code...");
+        //Good Notification
```
Comment (san983) [inline public/javascripts/controllers/widget.portal.js]: Why this comment?

File: public/javascripts/controllers/widget.portal.js [hunk 3]
```diff
@@ -149,11 +151,13 @@ define([
         .then(function successCallback(response) {
             $scope.widgetCode = response.data.widget_code;
             $scope.widget_form.$setPristine();
+            $scope.openNotice("success","Code Generated Successfully!");
+            // Good Notification.
```
Comment (san983) [inline public/javascripts/controllers/widget.portal.js]: Same as above, I think the `openNotice` here is self explanatory.

File: public/stylesheets/less/ngToast.less [hunk 4]
```diff
@@ -0,0 +1,176 @@
+
```
Comment (san983) [inline public/stylesheets/less/ngToast.less]: The indentation of this file is a mess, please check it.

File: public/config.js [hunk 5]
```diff
@@ -57,6 +62,9 @@ require.config({
       '//ajax.googleapis.com/ajax/libs/angularjs/1.4.12/angular-sanitize.min',
       '/lib/angular-sanitize/angular-sanitize.min'
     ],
+    'angular-toast': [
+      '/lib/ngtoast/dist/ngToast.min'
```
Comment (san983) [inline public/config.js]: Isn't there any CDN available with this JS library (to add a failover)? If not, that's ok, just asking

File: public/javascripts/controllers/widget.portal.js [hunk 6]
```diff
@@ -197,6 +201,14 @@ define([
       $scope.validAuthUri = false;
       $scope.invalidAuthUri = false;
     };
+    //NGToast Helper function.
+    $scope.openNotice = function (type, text) {
```
Comment (san983) [inline public/javascripts/controllers/widget.portal.js]: Indentation problem.

File: public/javascripts/controllers/widget.portal.js [hunk 7]
```diff
@@ -197,6 +201,14 @@ define([
...
+          ngToast.create({
+            className:type,
+            content: '<div>'+ text +"</div>"
+
```
Comment (san983) [inline public/javascripts/controllers/widget.portal.js]: why this empty line?

File: public/javascripts/controllers/widget.portal.js [hunk 8]
```diff
@@ -211,7 +223,9 @@ define([
     }
   };
 
-  WidgetEditController.$inject = ['$scope', '$http', '$window', '$controller', '$cookies', '$analytics'];
+
```
Comment (san983) [inline public/javascripts/controllers/widget.portal.js]: Same here, why these empty lines?

File: public/javascripts/controllers/widget.portal.js [hunk 9]
```diff
@@ -149,11 +151,13 @@ define([
...
             $scope.widgetCode = response.data.widget_code;
             $scope.widget_form.$setPristine();
+            $scope.openNotice("success","Code Generated Successfully!");
+            // Good Notification.
           },
           function errorCallback(response) {
             var data = response.data;
             console.log("Error: ", data);
-
+            $scope.openNotice("error","Error generating widget code snippet. Please check it.");
```
Comment (san983) [inline public/javascripts/controllers/widget.portal.js]: Check missing space after the first param please

File: public/config.js (hunk 5 above)
Comment (bnandaku) [inline public/config.js]: @san983  I have cleaned up the code, I was not able to find a good CDN for it right now but it is on my list. Working on tests.

Comment (san983) [inline public/config.js]: Cool. Once you have them we can call this complete 👍

Comment (bnandaku) [inline public/config.js]: @san983 Added the tests for toast. Please let me know if there is anything else I need to take care of

**Pull request merged**